File: exercism/python/bob/bob.py
def hey(what):
  what = what.strip()
  if what.isupper():
    return "Whoa, chill out!"
  elif what == "":
    return "Fine. Be that way!"
  elif what[-1] == '?':
    return "Sure."
  else:
    return "Whatever."

File: exercism/haskell/bob/src/Bob.hs
{-# LANGUAGE OverloadedStrings #-}

module Bob (responseFor) where

import Data.Char (isLower, isSpace, isUpper)
import qualified Data.Text as T

strip :: String -> String
strip = T.unpack . T.strip . T.pack

responseFor :: String -> String
responseFor s
  | all isSpace s = "Fine. Be that way!"
  | not (any isLower s) && any isUpper s = "Whoa, chill out!"
  | last (strip s) == '?' = "Sure."
  | otherwise = "Whatever."
